validator/client/runner.go

```diff
@@ -305,6 +306,33 @@ func runHealthCheckRoutine(ctx context.Context, v iface.Validator, eventsChan ch
 				return
 			}
 			isHealthy := tracker.CheckHealth(ctx)
+			if !isHealthy && len(hosts) > 1 && features.Get().EnableBeaconRESTApi {
+				for i, url := range hosts {
+					if url == v.RetrieveHost() {
+						next := (i + 1) % len(hosts)
+						log.Infof("Beacon node at %s is not responding, switching to %s", url, hosts[next])
+						v.UpdateHost(hosts[next])
+
+						km, err := v.Keymanager()
+						if err != nil {
+							log.WithError(err).Fatal("Could not get keymanager")
+						}
+						slot, err := v.CanonicalHeadSlot(ctx)
+						if err != nil {
+							log.WithError(err).Error("Could not get canonical head slot")
+							return
+						}
+						go func() {
```

Comment:
we don't need yet another gofunc here, the health check is already in a gofunc. just something to test.